src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -29,16 +33,825 @@ When you rebuild your Next.js site in Astro, you will notice some important diff
+      </div>
+    </>
+  )
+}
+
+export const getStaticPaths = async () => {
+  const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+  const resJson = await res.json();
+  const pokemons = resJson.results;
+
+  return {
+    paths: pokemons.map(({ name }) => ({
+      params: { name },
+    }))
+  }
+}
+
+export const getStaticProps = async (context) => {
+  const { name } = context.params
+  const [pokemon, species] = await Promise.all([
+    fetch(`https://pokeapi.co/api/v2/pokemon/${name}`).then(res => res.json()),
+    fetch(`https://pokeapi.co/api/v2/pokemon-species/${name}`).then(res => res.json())
+  ])
+
+  return {
+    props: {
+      pokemon: {
+        id: pokemon.id,
+        image: pokemon.sprites.front_default,
+        name: capitalize(pokemon.name),
+        height: pokemon.height,
+        weight: pokemon.weight,
+        flavorText: species.flavor_text_entries[0].flavor_text,
+        types: pokemon.types.map(({ type }) => type.name).join(', ')
+      },
+    },
+  }
+}
+```
+
+#### Create `src/pages/pokemon/[name].astro`
+
+Use the return value of the Next function. Convert any Next or React syntax to Astro.
+
+Note that:
+
+- Imported styles are moved to the code fence.
+- `className` becomes `class`.
+- `<Layout>` is added to provide the page shell.
+- `{pokemon.*}` values just work!
+
+```astro title="src/pages/pokemon/[name].astro" "class"
+---
+import styles from '../../styles/pokemon-entry.module.css';
+---
+
+<Layout>
+  <img class={styles.pokeImage} src={pokemon.image} alt={`${pokemon.name} picture`} />
+  <div class={styles.infoContainer}>
+    <h1 class={styles.header}>No. {pokemon.id}: {pokemon.name}</h1>
+    <table class={styles.pokeInfo}>
+      <tbody>
+        <tr>
+          <th>Types</th>
+          <td>{pokemon.types}</td>
+        </tr>
+        <tr>
+          <th>Height</th>
+          <td>{pokemon.height}</td>
+        </tr>
+        <tr>
+          <th>Weight</th>
+          <td>{pokemon.weight}</td>
+        </tr>
+      </tbody>
+    </table>
+    <p class={styles.flavor}>{pokemon.flavorText}</p>
+  </div>
+</Layout>
+```
+#### Add any needed imports, props and JavaScript
+
+
+1. Convert Next's `getStaticPaths()` function to Astro's `getStaticPaths()` by removing the `paths` route prefix and returning an array. (Otherwise, the functions are the same.) This function goes into Astro's code fence.
+
+:::tip
+Use `Astro.props` to access the `params` returned from the `getStaticPaths` function.
+:::
+
+```astro {4-11, 19} del={13-17} title="src/pages/pokemon/[name].astro"
+---
+import styles from '../../styles/pokemon-entry.module.css';
+
+export const getStaticPaths = async () => {
+  const res = await fetch("https://pokeapi.co/api/v2/pokemon?limit=151")
+  const resJson = await res.json();
+  const pokemons = resJson.results;
+
+  return pokemons.map(({ name }) => ({
+      params: { name },
+    }))
+
+  return {
+  paths: pokemons.map(({ name }) => ({
+    params: { name },
+  }))
```

Comment:
This is pretty hard to grok at first glance. Could we try replacing this with file tabs of Next -> Astro or even just having the full `getStaticPaths` for a before/after?

Reply:
resolving because we're not using this example in the final version :partying_face: